lib/linguist/heuristics.yml

```diff
@@ -226,7 +226,7 @@ disambiguations:
   - language: Erlang
     pattern: '^\s*(?:%%|main\s*\(.*?\)\s*->)'
   - language: JavaScript
-    pattern: '\/\/|("|'')use strict\1|export\s+default\s|\/\*(?:.|[\r\n])*?\*\/'
+    pattern: '\/\/|["'']use strict["'']|export\s+default\s|\/\*(?:.|[\r\n])*?\*\/'
```

Comment:
If this is a char-class, then it could be simplified to `["']`

Reply:
Yeah, the original regex was trying to account for the case with 2 single quotes, but in practice, that is not used (see this [search query](https://github.com/search?q=%2F%27%27use+strict%27%27%2F+path%3A*.es&type=code)). The goal of this PR wasn't to change the behavior of the regex, but it did in a small unconsequential way.

Now, we could either go all the was and make it a char-class or revert it back to `("|'')`, but it doesn't make a big difference.

Reply:
However, the previous rule using `("|'')` wasn't matching for the case where a unique single quote was used, which is the [most common occurence](https://github.com/search?q=%2F%27use+strict%27%2F+path%3A*.es&type=code) of `use script` in `.es` files, so `["']` is actually simpler and more in line with was appears in `.es` files.

@Rudxain, since the PR was already merged, feel free to open a new PR and make your suggested change to simplify the rules.

Reply:
Actually, you need the 2 single quotes inside a single-quoted yaml string, so what was merged is equivalent to `["']`, it's just in yaml notation.

Reply:
@DecimalTurn thanks for explaining!

Reply:
My pleasure. It was also a good reminder for me as well 😅